client/modules/food/components/inventory/FoodItems.js

```diff
@@ -146,6 +150,27 @@ class FoodItems extends React.Component {
     this.setState({ validInput: this.getValidationState.all() && this.checkChanged() })
   }
 
+  touch = {
+    foodName: () => this.setState({ touched: { ...this.state.touched, foodName: true } }, this.validate),
+    foodCategory: () => this.setState({ touched: { ...this.state.touched, foodCategory: true } }, this.validate),
+    foodQuantity: () => this.setState({ touched: { ...this.state.touched, foodQuantity: true } }, this.validate),
+  }
+
+  /**
+   *  callback for setState when user selects an existing food name from Autosuggest
+   */
+   validateFocus = () => {
+     if(this.validate){
```

Comment:
`this.validate` is a function though, so it'll always be truthy. Looking at it more, `validateFocus` is only called after setting state to an existing food item, so it'll always be valid anyway and checking isn't necessary. 

It's a bit weird if you type a complete existing food name, the dropdown stays open and focus moves to quantity. The `closeOnCompletion` prop doesn't seem to help.

Reply:
Ahh, good catch on this.Validate always being truthy, can't believe I missed the point of that 🤦 .  

I'll take another stab at this tonight/tomorrow.

Reply:
Cool, I don't know if it's worth spending much time on though, seems a bit of an edge case.

Reply:
Got it cleaned up a bit, removed the ValidateFocus since it's largely unnecessary, the only thing lost is moving the cursor for the case where someone edits an item and uses autosuggest to select a new item.

I think the weird behavior with Autosuggest and the `closeOnCompletion` is an internal problem with Autosuggest, I couldn't get that prop to work even outside this project in a simple setting.

I did find a react warning when you open the autosuggest list when more than one item has the same name.  It can be fixed by changing `itemReactKeyPropName` to id instead of name.  Wasn't sure if I should open an issues for something so trivial or just fix it here?

> Warning: flattenChildren(...): Encountered two children with the same key, `.0:$Peas`. Child keys must be unique; when two children share a key, only the first child will be used.